plugins/wasi_nn/neuralspeed.cpp

```diff
@@ -0,0 +1,338 @@
+    Py_XDECREF(GraphRef.Model);
+    Py_XDECREF(GraphRef.ModelClass);
+    Py_XDECREF(GraphRef.NeuralSpeedModule);
+    return WASINN::ErrNo::InvalidArgument;
+  }
+  Py_XDECREF(LoadResult);
+  GraphRef.LoadTime = std::chrono::duration_cast<std::chrono::milliseconds>(
+                          std::chrono::steady_clock::now() - StartTime)
+                          .count();
+  return WASINN::ErrNo::Success;
+}
+
+Expect<WASINN::ErrNo> initExecCtx(WasiNNEnvironment &Env, uint32_t GraphId,
+                                  uint32_t &ContextId) noexcept {
+  if (!Py_IsInitialized()) {
+    spdlog::info(
+        "[WASI-NN][Error] Neural speed backend: Model has been realse, please reload it."sv);
+    return WASINN::ErrNo::RuntimeError;
+  }
+  Env.NNContext.emplace_back(GraphId, Env.NNGraph[GraphId]);
+  ContextId = Env.NNContext.size() - 1;
+  return ErrNo::Success;
+}
+
+Expect<WASINN::ErrNo> setInput(WasiNNEnvironment &Env, uint32_t ContextId,
+                               uint32_t, const TensorData &Tensor) noexcept {
+  auto &CxtRef = Env.NNContext[ContextId].get<Context>();
+  auto &GraphRef = Env.NNGraph[CxtRef.GraphId].get<Graph>();
+  if (!Py_IsInitialized()) {
+    spdlog::info(
+        "[WASI-NN][Error] Neural speed backend: Model has been realse, please reload it."sv);
+    return WASINN::ErrNo::RuntimeError;
+  }
+  if (GraphRef.EnableDebugLog) {
+    spdlog::info("[WASI-NN][Debug] Neural speed backend: setInput"sv);
+  }
+
+  // Set the input.
+  if (Tensor.Tensor.size() % sizeof(long long int) != 0) {
+    spdlog::error("[WASI-NN] neural speed backend: Input tensor size is not a "
+                  "multiple of "
+                  "4 bytes."sv);
+    return WASINN::ErrNo::InvalidArgument;
+  }
+  const std::vector<long long int> Prompt{
+      reinterpret_cast<const long int *>(Tensor.Tensor.data()),
+      reinterpret_cast<const long *>(Tensor.Tensor.data() +
+                                     Tensor.Tensor.size())};
+  CxtRef.Inputs.clear();
+  CxtRef.Inputs = Prompt;
+
+  return WASINN::ErrNo::Success;
+}
+Expect<WASINN::ErrNo> getOutput(WasiNNEnvironment &Env, uint32_t ContextId,
+                                uint32_t, Span<uint8_t> OutBuffer,
+                                uint32_t &BytesWritten) noexcept {
+  auto &CxtRef = Env.NNContext[ContextId].get<Context>();
+  auto &GraphRef = Env.NNGraph[CxtRef.GraphId].get<Graph>();
+  if (GraphRef.EnableDebugLog) {
+    spdlog::info("[WASI-NN][Debug] Neural speed backend: getOutput"sv);
+  }
+  std::string StringTmp(reinterpret_cast<const char *>(CxtRef.Outputs.data()),
+                        CxtRef.Outputs.size() * sizeof(long long int));
+  std::copy_n(StringTmp.data(), StringTmp.length(), OutBuffer.data());
+  BytesWritten = StringTmp.length();
+  return WASINN::ErrNo::Success;
+}
+Expect<WASINN::ErrNo> compute(WasiNNEnvironment &Env,
+                              uint32_t ContextId) noexcept {
+  if (!Py_IsInitialized()) {
+    spdlog::info(
```

Comment:
We should use `spdlog::error` here. This is a fatal error.

Reply:
I have changed. Please check, thank you.